app/lib/utils.ts

```diff
@@ -31,3 +36,14 @@ export function getFormDataString(formData: FormData, key: string) {
     throw new Response(`expected ${key} to be a string`, { status: 400 })
   }
 }
+
+export async function refreshUser(
+  user: NonNullable<Awaited<ReturnType<typeof getUser>>>
+) {
+  const client = await getOpenIDClient()
+  const refreshedTokenSet = await client.refresh(user.refreshToken)
+  const user_new = userFromTokenSet(refreshedTokenSet)
+  await updateSession(user_new)
+  user.groups = user_new.groups
+  user.token = user_new.token
+}
```

Comment:
I thought that JS is generally pass-by-value, not pass-by-reference. Does this actually update the user?

Reply:
Yea, it works. I threw a console log to check, but if i go into cognito and change the groups and just call the refresh user method I get the new groups 
![image](https://user-images.githubusercontent.com/11023698/203401913-7523fbc1-9213-4952-92d5-5909c8f06360.png)